client/src/components/MemberList.js

```diff
@@ -8,46 +9,65 @@ import TableHead from '@material-ui/core/TableHead';
+function createMemberData(id, username, gender, age, status, budget) {
+    return { id, username, gender, age, status, budget };
 }
 
-var rows = [];
-
 export default function MemberList(props) {
-    //window.alert(props.data);
-    if(props.data !== undefined) {
-        rows = props.data;
+    const [state, setState] = useState(0);
+    // if (state > 0) {
+    //     isClicked();
+    //     //rows = props.data;
+    // }
+    // if (state == 0) {
+    //     isClicked2();
+    //     //rows = props.data;
+    // }
+    if (props.data !== undefined) {
+        if(rows !== props.data) {
+            rows = props.data;
+            setState(0);
+        }
+        
     }
     const classes = useStyles();
     return (
-        <TableContainer component={Paper}>
-            <Table className={classes.table} size="small" aria-label="a dense table">
-                <TableHead>
-                    <TableRow>
-                        <TableCell>User</TableCell>
-                        <TableCell align="right">Gender</TableCell>
-                        <TableCell align="right">Age</TableCell>
-                    </TableRow>
-                </TableHead>
-                <TableBody>
-                    {rows.map((row) => (
-                        <TableRow key={row.id}>
-                            <TableCell component="th" scope="row">
-                                {row.username}
-                            </TableCell>
-                            <TableCell align="right">{row.gender}</TableCell>
-                            <TableCell align="right">{row.age}</TableCell>
+        <div>
+            <TableContainer component={Paper}>
+                <Table className={classes.table} size="small" aria-label="a dense table">
+                    <TableHead>
+                        <TableRow>
+                            <TableCell>User</TableCell>
+                            <TableCell align="right">Gender</TableCell>
+                            <TableCell align="right">Age</TableCell>
+                            <TableCell align="right">Status</TableCell>
+                            <TableCell align="right">Budget</TableCell>
                         </TableRow>
-                    ))}
-                </TableBody>
-            </Table>
-        </TableContainer>
+                    </TableHead>
+                    <TableBody>
+                        {rows.map((row) => (
+                            <TableRow key={row.id}>
+                                <TableCell component="th" scope="row">
+                                    {row.username}
+                                </TableCell>
+                                <TableCell align="right">{row.gender}</TableCell>
+                                <TableCell align="right">{row.age}</TableCell>
+                                <TableCell align="right">{row.status}</TableCell>
+                                <TableCell align="right">{row.budget}</TableCell>
```

Comment:
Just use `props.data` instead of `row` here. We don't need to (and shouldn't) mess with global variables or state for this usecase.

Also, I would rename `props.data` to `props.members` so the context makes sense within this component. Then you could say for example:
```js
props.members.map((member) => (
```